components/member_list_team/member_list_team.tsx

```diff
@@ -1,36 +1,48 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
+import {getTeamStats} from 'mattermost-redux/actions/teams';
+import {searchProfiles} from 'mattermost-redux/actions/users';
+
 import Constants from 'utils/constants';
 import * as UserAgent from 'utils/user_agent';
 
 import SearchableUserList from 'components/searchable_user_list/searchable_user_list_container.jsx';
 import TeamMembersDropdown from 'components/team_members_dropdown';
 
+import {loadStatusesForProfilesList} from 'actions/status_actions.jsx';
+import {loadProfilesAndTeamMembers, loadTeamMembersForProfilesList} from 'actions/user_actions.jsx';
+import {setModalSearchTerm} from 'actions/views/search';
+
 const USERS_PER_PAGE = 50;
 
-export default class MemberListTeam extends React.Component {
-    static propTypes = {
-        searchTerm: PropTypes.string.isRequired,
-        users: PropTypes.arrayOf(PropTypes.object).isRequired,
-        teamMembers: PropTypes.object.isRequired,
-        currentTeamId: PropTypes.string.isRequired,
-        totalTeamMembers: PropTypes.number.isRequired,
-        canManageTeamMembers: PropTypes.bool,
-        actions: PropTypes.shape({
-            searchProfiles: PropTypes.func.isRequired,
-            getTeamStats: PropTypes.func.isRequired,
-            loadProfilesAndTeamMembers: PropTypes.func.isRequired,
-            loadStatusesForProfilesList: PropTypes.func.isRequired,
-            loadTeamMembersForProfilesList: PropTypes.func.isRequired,
-            setModalSearchTerm: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    searchTerm: string;
+    users: Array<any>;
+    teamMembers: any;
+    currentTeamId: string;
+    totalTeamMembers: number;
+    canManageTeamMembers?: boolean;
+    actions: {
+        searchProfiles: typeof searchProfiles;
```

Comment:
Not blocking, but I'm wondering if we should be tying the prop types directly to `mattermost-redux`. As much as it looks cleaner, it reduces modularity.

Reply:
After toying around with different ways to do this, this was the best working method in my opinion. The problem with redefining the function here is that, right now, not all these functions are typed as expected. Long story short, the return value does not have in some cases the "then" function. If we change that, it does not work with the binding done in index because the type does not match.

A solution would be to define the function as (args) => any, but in my opinion that is not typescript. I have been struggling with that opinion in many parts of this migration because I had to use 'any' in more places I would like, but if you have any suggestion, I am more than willing to dig deeper into that.

Reply:
Okay, I'm happy with this then. Thanks for providing insight :)

Reply:
If those functions have inconsistent return types, then we should wrap some of those return statements in `Promise.resolve` so that they're all consistent.

I don't really like coupling these props to the related actions since you should ideally be able to use these components without the corresponding actions. I'd prefer to use the actual types here if possible.

Reply:
After a discussion with @jespino I realised I had a problem with `bindActionCreators`. I will upload a new commit soon that will solve many of the issues.